### ISSUE TITLE
Roundabout exit is always returned 2nd

### ISSUE THREAD

**Issue by safe-bug** [valhalla/valhalla#2320]
Hello
I've built and used the latest release of Valhalla (3.0.9), but when I get routings that have roundabouts in, the exit number is always return the 2nd. I also tried the master and previous releases and the issue is still remaining. (Only the demo page of Valhalla returns the correct exit)

How can I fix it?

**Comment by kevinkreiser** [comment]
this may be caused by building tiles without an admin database (which determines whether your country in question drives on the left or right side of the road). the best way to fix this is download the planet and run `valhalla_build_admins` and then provide the resulting sqlite database to `valhalla_build_tiles` whenever you want to build tiles.

@gknisely @dgearhart what do you guys think?

**Comment by dgearhart** [comment]
@safe-bug yes, we have seen this issue if the driving side of street is not correct

**Comment by safe-bug** [comment]
@kevinkreiser @dgearhart I built admins.sqlite with my country pbf (downloaded from osm) and tried to build routing tiles with my own osm file and the admins.sqlite, but the result is the same as previous.

Should my own osm file also contain some info that exists on the planet?

**Comment by dgearhart** [comment]
@gknisely do you have more info/advice? I verified the roundabout exit count is correct with latest code and data

**Comment by gknisely** [comment]
@safe-bug are you sure you are using the admins.sqlite?  Can you check your config and make sure it points to the correct file.

**Comment by kevinkreiser** [comment]
i didnt emphasize it but i did say planet before for a particular reason. it is possible that your contries extract doesnt have the proper relations to build the admin.sqlite. it depends on how the data was extracted from the planet file. when all else fails, make the admin.sqlite from the planet pbf to be sure you have all the country border relations and that none have been clipped.

**Comment by safe-bug** [comment]
@dgearhart @gknisely @kevinkreiser 
I tried to build admins.sqlite for my county pbf, but the resulting routing graph had the same issue with roundabout exits. But when I used the planet pbf to build admins, the problem has solved!

I think when we don't set any config for the "side of driving", Valhalla uses the right hand as a default, so why it makes a problem for roundabouts? And if admins.sqlite absence is gonna make some problems like this, I think it would be better to add some kind of warning here in [Running](https://github.com/valhalla/valhalla#running) section of the Readme file to notify that it is a required file. 

Thanks for your help guys.

**Comment by st4141** [comment]
As @safe-bug wrote, why can't Valhalla use "driving on right side" by default and remove this problem of 2nd exit on roundabout ?
Instead of downloading only ile-de-france-latest.osm.pbf (235Mb) to generate a valhalla_tiles.tar that is 150Mb big, I have to download the planet (55Gb) + ile-de-france-latest.osm.pbf to generate a valhalla_tiles.tar that is 500Mb big because it includes a admins.sqlite from World that is 360Mb by itself)

**Comment by dnesbitt61** [comment]
Why this cannot be updated: there is a 1 bit flag in the data within NodeInfo::drive_on_right_. This flag is true if driving is on the right hand side. This cannot be changed without a major version update which means anyone updating code needs to update data. To keep data compatible and allow use of older data with new code this needs to stay as is.

**Comment by nilsnolde** [comment]
@dnesbitt61 would it make sense for you to create some container (milestone, repo project) to add things like this to for a (probably fairly far away) future major update? Backwards-incompatible changes which could be useful/sensible might be worth keeping track of?

**Comment by kevinkreiser** [comment]
@nilsnolde there are a couple ideas kicking around like that, its not a bad idea. i have made at least one branch with the term `v4` (as in next tile version) in it which would fit under this umbrella

**Comment by nilsnolde** [comment]
Github's kanban-style boards are IMO pretty useful for that sort of stuff. Columns could be set up by priority or even different boards/projects for different parts of valhalla. It's easier to communicate when approaching a major cycle. Still, it's some additional maintenance..

**Comment by st4141** [comment]
I made some tests and realized that admis.sqlite doesn't need to be in valhalla_tiles.tar.
Good news ! I'll keep generating "small" files just by adding a grep -v or a filter to find to the command line used to generate valhalla_tiles.tar
find valhalla_tiles | sort -n | tar cf valhalla_tiles.tar --no-recursion -T -

**Comment by st4141** [comment]
Me again.
From ile-de-france-latest.osm.pbf (a region of France) I generated admins.sqlite then the tiles and the tar file.
Without changing anything, **I have the "2nd exit" bug**.
The generated table admins (in admins.sqlite) contains only 1 record.
In this record:
- **admin_level = 4**
- drive_on_right = 1
- geom contains a blob (I guess this is the drawing of the region)
If I change **admin_level to 2** and generate the tiles again, **I don't have the "2nd exit" bug anymore**.

Is Valhalla looking for the driving side in this table only for record where admin_level = 2 ? (Or 0 to 3 but not 4 ?)
